### PR TITLE
RI-5036: Remove mdns, avahi, gnome-keyring; use alpine image instead …

### PR DESCRIPTION
**Pull request by nelsonwellsredis**
…of slim

### CODE REVIEW

File: docker-entry.sh [hunk 1]
```diff
@@ -7,18 +7,8 @@
...
-
-# Launching system's secret storage
-eval "$(dbus-launch --sh-syntax)"
-mkdir -p ~/.cache
-mkdir -p ~/.local/share/keyrings # where the automatic keyring is created
-eval "$(echo "$GNOME_KEYRING_PASS" | gnome-keyring-daemon --unlock)"
-sleep 1
-eval "$(echo "$GNOME_KEYRING_PASS" | gnome-keyring-daemon --start)"
+echo "Running docker-entry.sh as:"
+whoami
```
Comment (arthosofteq) [inline docker-entry.sh:11]: I am "node" :)

**Review by arthosofteq — COMMENTED** [review]
Do you plan any optimizations in scope of this task?

File: docker-entry.sh (hunk 1 above)
Comment (vlad-dargel) [reply docker-entry.sh:11]: also e2e tests fail with different reasons:
1. Looks like Encryption doesn't work - its option in settings is disabled
2. Something with Namespaces sorting on Database analysis page
3. Profiler logs not saved
4. RE Cluster doesn't work
5. SQLITE_ERROR: no such table: features_config when test trying to find table in db
6. Wrong number of imported connections test. Looks like some connections were failed
7. TLS certificates empty on connections import. The same is with connections with SSH.
8. New published notification not displayed
9. EACCES: permission denied. mkdir 'tmp/RedisInsight-v2' when uploading custom tutorials

**Review by arthosofteq — APPROVED** [review]
lgtm

File: redisinsight/api/src/main.ts [hunk 2]
```diff
@@ -33,10 +34,7 @@ export default async function bootstrap(): Promise<IApp> {
   };
 
   if (serverConfig.tls && serverConfig.tlsCert && serverConfig.tlsKey) {
```
[the remark this answers is not in the document]
Comment (nelsonwellsredis) [reply redisinsight/api/src/main.ts:36]: Sounds good, I think I missed that point if we had discussed it earlier. I do like that better.

File: configs/webpack.config.main.prod.ts [hunk 3]
```diff
@@ -56,30 +57,61 @@ export default merge(baseConfig, {
       DEBUG_PROD: false,
       START_MINIMIZED: false,
       APP_ENV: 'electron',
-      SERVER_TLS: true,
-      SERVER_TLS_CERT: process.env.SERVER_TLS_CERT || '',
-      SERVER_TLS_KEY: process.env.SERVER_TLS_KEY || '',
+      RI_SERVER_TLS: true,
```
[the remark this answers is not in the document]
Comment (nelsonwellsredis) [reply configs/webpack.config.main.prod.ts]: Removed

File: redisinsight/api/config/default.ts [hunk 4]
```diff
@@ -42,7 +44,7 @@ export default {
     version,
     env: 'development',
     listenInterface: process.env.RI_HOSTNAME ?? '0.0.0.0',
-    port: 5000,
+    port: process.env.RI_APP_PORT ?? 5000,
```
[the remark this answers is not in the document]
Comment (nelsonwellsredis) [reply redisinsight/api/config/default.ts]: Fixed.

File: redisinsight/api/src/modules/server/server.service.ts [hunk 5]
```diff
@@ -64,7 +67,7 @@ export class ServerService implements OnApplicationBootstrap {
           appVersion: SERVER_CONFIG.appVersion,
           osPlatform: process.platform,
           buildType: SERVER_CONFIG.buildType,
-          port: process.env.API_PORT || SERVER_CONFIG.port,
+          port: process.env.RI_APP_PORT || SERVER_CONFIG.port,
```
[the remark this answers is not in the document]
Comment (nelsonwellsredis) [reply redisinsight/api/src/modules/server/server.service.ts]: Fixed.

File: redisinsight/api/src/main.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (nelsonwellsredis) [reply redisinsight/api/src/main.ts]: Fixed.

File: configs/webpack.config.web.prod.ts [hunk 6]
```diff
@@ -76,11 +76,11 @@ const configuration: webpack.Configuration = {
     new webpack.EnvironmentPlugin({
       NODE_ENV: 'production',
       APP_ENV: 'web',
-      API_PORT: '5000',
+      RI_APP_PORT: '5000',
       API_PREFIX: '',
       BASE_API_URL: 'api/',
       RESOURCES_BASE_URL:
-        process.env.SERVER_TLS_CERT && process.env.SERVER_TLS_KEY ? 'https://localhost' : 'http://localhost',
+        process.env.RI_SERVER_TLS_CERT && process.env.RI_SERVER_TLS_KEY ? 'https://localhost' : 'http://localhost',
```
Comment (arthosofteq) [inline configs/webpack.config.web.prod.ts:83]: config.web.prod used for web. I thought that we shouldn't rely on tls and UI should have generic implementation.
@romansergeenkosofteq @zalenskiSofteq Could you confirm that we can remove RESOURCES_BASE_URL env?

File: redisinsight/api/config/default.ts [hunk 7]
```diff
@@ -41,8 +41,8 @@ export default {
   server: {
     version,
     env: 'development',
-    listenInterface: process.env.RI_HOSTNAME ?? '0.0.0.0',
-    port: 5000,
+    host: process.env.RI_APP_HOST ?? '0.0.0.0',
+    port: parseInt(process.env.RI_APP_PORT, 10) ?? 5000,
```
Comment (arthosofteq) [inline redisinsight/api/config/default.ts:45]: When RI_APP_PORT is not specified server.port will be NaN but not 5000

File: redisinsight/api/src/utils/config.spec.ts [hunk 8]
```diff
@@ -24,7 +24,7 @@ describe('Config util', () => {
       // eslint-disable-next-line global-require
       const { get } = require('./config');
 
-      const result = get('server');
+      const result = get('server') as Config['server'];
```
Comment (arthosofteq) [inline redisinsight/api/src/utils/config.spec.ts:27]: There is no import of "Config" in the top of the file.

File: configs/webpack.config.web.prod.ts (hunk 6 above)
Comment (zalenskiSofteq) [reply configs/webpack.config.web.prod.ts:83]: no problem

Comment (arthosofteq) [reply configs/webpack.config.web.prod.ts:83]: @nelsonwellsredis Could you please remove then?

**Review by AmirAllayarovSofteq — APPROVED** [review]

**Review by zalenskiSofteq — APPROVED** [review]

**Pull request merged**